From 877b07aa1d87624a2929e4e1aa792c848f42f173 Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@streambits.io>
Date: Mon, 13 Jan 2020 13:25:42 +0200
Subject: [PATCH 03/14] gpt-boot-generator: Do not automount the ESP on Solus.

We use clr-boot-manager, thus it is undesirable to automount the ESP, as we want it unmounted constantly. When required, clr-boot-manager will locate the ESP, mount it, perform updates, and unmount it again.
---
 src/gpt-auto-generator/gpt-auto-generator.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/gpt-auto-generator/gpt-auto-generator.c b/src/gpt-auto-generator/gpt-auto-generator.c
index cc5fc3f813..30f13c9cef 100644
--- a/src/gpt-auto-generator/gpt-auto-generator.c
+++ b/src/gpt-auto-generator/gpt-auto-generator.c
@@ -574,6 +574,9 @@ static int add_partition_esp(DissectedPartition *p, bool has_xbootldr) {
                 return 0;
         }
 
+        log_debug("/boot automount is disabled on Solus, exiting.");
+        return 0;
+
         /* Check if there's an existing fstab entry for ESP. If so, we just skip the gpt-auto logic. */
         r = fstab_has_node(p->node);
         if (r < 0)
-- 
2.43.0

